spark/processing/3.1/py3/docker/py39/Dockerfile.cpu

```diff
@@ -3,20 +3,33 @@ ARG REGION
 ENV AWS_REGION ${REGION}
 RUN yum clean all
 RUN yum update -y
-RUN yum install -y awscli bigtop-utils curl gcc gzip unzip python3-setuptools python3-pip python-devel python3-devel python-psutil gunzip tar wget liblapack* libblas* libopencv* libopenblas*
+RUN yum install -y awscli bigtop-utils curl gcc gzip unzip gunzip tar wget liblapack* libblas* libopencv* libopenblas*
+
 
 # Install python 3.9
-RUN yum -y install gcc openssl-devel bzip2-devel libffi-devel
-RUN yum -y install tar xz wget gzip make
-RUN wget https://www.python.org/ftp/python/3.9.9/Python-3.9.9.tgz
-RUN tar xzf Python-3.9.9.tgz
-RUN cd Python-*/ && ./configure --enable-optimizations && make altinstall
-RUN echo 'alias python3=python3.9' >> ~/.bashrc
+ARG PYTHON_BASE_VERSION=3.9
+ARG PYTHON_WITH_BASE_VERSION=python${PYTHON_BASE_VERSION}
+ARG PIP_WITH_BASE_VERSION=pip${PYTHON_BASE_VERSION}
+
+ARG PYTHON_VERSION="3.9.12"
+RUN yum -y groupinstall 'Development Tools' \
+    && yum -y install openssl-devel bzip2-devel libffi-devel sqlite-devel xz-devel \
+    && wget https://www.python.org/ftp/python/${PYTHON_VERSION}/Python-${PYTHON_VERSION}.tgz \
+    && tar xzf Python-${PYTHON_VERSION}.tgz \
+    && cd Python-*/ \
+    && ./configure --enable-optimizations \
+    && make altinstall \
+    && echo 'alias python3=python3.9\nalias pip3=pip3.9' >> ~/.bashrc \
+    && ln -s $(which ${PYTHON_WITH_BASE_VERSION}) /usr/local/bin/python3 \
+    && ln -s $(which ${PIP_WITH_BASE_VERSION}) /usr/local/bin/pip3
+
+# clean up python
+RUN rm Python-${PYTHON_VERSION}.tgz
+RUN rm -rf Python-${PYTHON_VERSION}
```

Comment:
These extra RUN commands will increase the image size, and `rm` won't decrease it as the image layer is still there.  If you want `rm` them, I suggest doing it in the same RUN command above

Reply:
Also, don't you need to `cd ..` before removing?